ext/dom/element.c

```diff
@@ -296,6 +297,7 @@ PHP_METHOD(DOMElement, setAttribute)
 		attr = (xmlNodePtr)xmlSetProp(nodep, (xmlChar *) name, (xmlChar *)value);
 	}
 	if (!attr) {
+		// TODO Convert to error?
```

Comment:
tricky, why should xmlSetProp fail though, there isn't even a test for this. I would say go for it.